### PR TITLE
feat: node 18 support, crawlee and bullseye

### PR DESCRIPTION
**Pull request by vladfrangu**
- Adds images for nodejs 18
- Upgrades all images that use debian to debian 11 bullseye (also means latest playwright works in them natively)
- updated all the test code to crawlee as well as adds it as a dependency to all images
- Closes #79

### CODE REVIEW

**Comment by vladfrangu** [comment]
Right off the bat I can see that nodejs 14 and 15 are probably to be removed if we want to move to actor sdk + crawlee... Is that ok? Are we ready for that?

**Comment by szmarczak** [comment]
> Right off the bat I can see that nodejs 14 and 15 are probably to be removed if we want to move to actor sdk + crawlee... Is that ok? Are we ready for that?

I'd wait once people start using Node.js 18 more (a few weeks). Then there will be a drop-off in Node.js 14 users.

File: node-playwright-webkit/Dockerfile [hunk 1]
```diff
@@ -38,6 +35,11 @@ RUN npm config --global set update-notifier false
 # Install all required playwright dependencies for webkit
 RUN PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=1 npm_config_ignore_scripts=1 npx playwright install-deps webkit
 
+# Cleanup apt caches
+RUN apt-get clean -y && apt-get autoremove -y \
+    && rm -rf /var/lib/apt/lists/* \
+    && rm -rf /src/*.deb
+
```
Comment (szmarczak) [inline node-playwright-webkit/Dockerfile:42]: Is this required?

**Review by szmarczak — APPROVED** [review]

File: node-playwright-webkit/Dockerfile (hunk 1 above)
Comment (vladfrangu) [reply node-playwright-webkit/Dockerfile:42]: It's something we've done in all images to shave off some extra space from the redundant deb packages cached by apt (that said I never tried it with/without it so...maybe it's not? I could try it out and see if theres a size diff)

**Comment by B4nan** [comment]
I would drop node < 16 right ahead. Why should we base this on usage of node 18? Also, the old images will be still there for people using v2. Using v3 requires node 16.

**Comment by pocesar** [comment]
this will also instantly fix this: https://github.com/nodejs/node/issues/39400 and https://github.com/nodejs/node/issues/32978

**Comment by B4nan** [comment]
> this will also instantly fix this: https://github.com/nodejs/node/issues/39400 and https://github.com/nodejs/node/issues/32978

fyi node 18 images will be opt in before lts is out, we will keep node 16 in the templates

**Review by mnmkng — APPROVED** [review]
LGTM. Please test how template build times look on the platform with these images. The target is < 1 min.

**Comment by pocesar** [comment]
the hardcoded `NODE_VERSION=16` will make it possible to use `:18` on Dockerfile? or is there a separate image that will be created?

File: node/package.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
         "start": "node main.js"
     },
     "dependencies": {
-        "apify": "APIFY_VERSION"
+        "apify": "APIFY_VERSION",
```
Comment (B4nan) [inline node/package.json:10]: this should also have `crawlee` installed, or not?

**Review by B4nan — APPROVED** [review]

**Comment by vladfrangu** [comment]
> the hardcoded `NODE_VERSION=16` will make it possible to use `:18` on Dockerfile? or is there a separate image that will be created?

Yep! That's just the default value if not specified! If we specify NODE_VERSION=18 in CI we'll build a node 18 image

**Pull request merged**